### PR TITLE
[DCOS-57182] Release Data Science Engine 1.0.0 GA

### PR DESCRIPTION
**Pull request by akirillov**
## Release Data Science Engine 1.0.0 GA
Source: https://github.com/mesosphere/jupyter-service/releases/tag/1.0.0

### CODE REVIEW

File: repo/packages/D/data-science-engine/0/config.json [hunk 1]
```diff
@@ -0,0 +1,822 @@
...
+              "description": "Number of Nvidia GPUs to allocate (e.g., 2) to this service (Data Science Engine) instance. Note: Nvidia GPU Support must be enabled for this to take effect.",
+              "minimum": 0,
+              "default": 0
+            }
+          }
+        },
+        "jupyter_password": {
+          "type": "string",
+          "title": "Data Science Engine Notebook Password",
+          "description": "Data Science Engine Notebook Password (JUPYTER_PASSWORD). Defaults to jupyter unless OpenID Connect has been configured.",
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/config.json]:
```suggestion
          "description": "Data Science Engine Notebook Password (JUPYTER_PASSWORD). Defaults to 'jupyter' unless OpenID Connect has been configured.",
```

**Review by takirala — APPROVED** [review]

File: repo/packages/D/data-science-engine/0/marathon.json.mustache [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+        {{#security.tls.jaas_secret}}
+          ,{
+            "containerPath": "secrets/jaas.conf",
+            "secret": "tls_jaas"
+          }
+        {{/security.tls.jaas_secret}}
+      {{/security.tls.enabled}}
+    ],
+    "docker": {
+
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/marathon.json.mustache]:
```suggestion
```

File: repo/packages/D/data-science-engine/0/marathon.json.mustache [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+        {{^advanced.jupyter_notebook_image}}
+            "image": "{{resource.assets.container.docker.jupyter-service-notebook-image-gpu-cuda100}}",
+        {{/advanced.jupyter_notebook_image}}
+      {{/service.gpu.enabled}}
+      {{^service.gpu.enabled}}
+        "image": "{{resource.assets.container.docker.jupyter-service-notebook-image-cpu}}",
+      {{/service.gpu.enabled}}
+      "forcePullImage": {{advanced.force_pull_jupyter_image}}
+    }
+
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/marathon.json.mustache]:
```suggestion
```

File: repo/packages/D/data-science-engine/0/marathon.json.mustache [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+          "containerPort": 18080,
+          "protocol": "tcp",
+          "name": "sparkhistory",
+          "labels": {
+            "VIP_8": "{{service.name}}:18080"
+          }
+        }
+    ]
+  {{/networking.virtual_network_enabled}}
+
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/marathon.json.mustache]:
```suggestion
```

File: repo/packages/D/data-science-engine/0/marathon.json.mustache [hunk 5]
```diff
@@ -0,0 +1,644 @@
...
+  },
+
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/marathon.json.mustache]:
```suggestion
```

File: repo/packages/D/data-science-engine/0/marathon.json.mustache [hunk 6]
```diff
@@ -0,0 +1,644 @@
...
+      "port": 0,
+      "protocol": "tcp",
+      "name": "sparkhistory",
+      "labels": {
+        "VIP_7": "{{service.name}}:18080"
+      }
+    }
+  ],
+  {{/networking.virtual_network_enabled}}
+
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/marathon.json.mustache]:
```suggestion
```

File: repo/packages/D/data-science-engine/0/marathon.json.mustache [hunk 7]
```diff
@@ -0,0 +1,644 @@
...
+        "source": "{{storage.s3.aws_access_key_id}}"
+     }
+    {{/storage.s3.aws_access_key_id}}
+    {{#storage.s3.aws_secret_access_key}}
+    ,"aws_secret_access_key": {
+        "source": "{{storage.s3.aws_secret_access_key}}"
+     }
+    {{/storage.s3.aws_secret_access_key}}
+  },
+
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/marathon.json.mustache]:
```suggestion
```

File: repo/packages/D/data-science-engine/0/package.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    "jupyter",
+    "jupyterlab",
+    "notebook",
+    "openid-connect",
+    "ray",
+    "s3",
+    "spark",
+    "tensorflow"
+  ],
+  "preInstallNotes": "Default configuration requires 1 agent node with: 1 CPU | 4096 MB MEM",
```
Comment (samvantran) [inline repo/packages/D/data-science-engine/0/package.json:27]: Should we add a note about HDFS requiring an additional 5 nodes?

**Review by samvantran — APPROVED** [review]
Looks good @akirillov! Minor suggestions my side

File: repo/packages/D/data-science-engine/0/package.json (hunk 8 above)
Comment (akirillov) [reply repo/packages/D/data-science-engine/0/package.json:27]: I don’t think it’s relevant because this requirement is stated in HDFS pre-install notes

**Comment by akirillov** [comment]
Thanks for the review @samvantran, added fixes to all the comments.

**Pull request merged**